app/src/main/java/com/github/orkest/ui/NavigationBar.kt

```diff
@@ -94,9 +96,16 @@ class NavigationBar {
                     composable("HomePage") { FeedActivity(PostViewModel()) }
                     composable("SearchPage") { SearchUserView.SearchUi(viewModel = viewModel) }
                     composable("ShazamPage") {
-                        ShazamSong(activity)
-                        val intent = Intent(context, CameraView::class.java)
-                        context.startActivity(intent)
+                        if(FireStoreDatabaseAPI.isOnline(context))
+                        {
+                            ShazamSong(activity)
+                            val intent = Intent(context, CameraView::class.java)
+                            context.startActivity(intent)
+                        }
+                        else{
+                            Toast.makeText(context, "No internet connection. Unable to post and shazam.", Toast.LENGTH_LONG).show()
```

Comment:
Actually you can shazam offline since the databse is loaded in local, only the posting won't be able to be done on the spot.

Reply:
I agree that that would have been a better behavior but there where other important tasks to finish and if offline functionality has a correct behavior than for the moment it's ok and I will improve it next week